sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -155,7 +156,7 @@ abstract class PercentileBase
     }
 
     val ordering = child.dataType match {
```

Comment:
how about we put `ordering` in the base `PhysicalDataType`? then we can just do `val ordering = PhysicalDataType(child.dataType).ordering`

Reply:
Why not use OrderedPhysicalDataType for this?

Reply:
What's the benefit of having `OrderedPhysicalDataType` and complicating the hierarchy of `PhysicalDataType`? Most callers just patter match some data types and fail for unexpected data types. It's simpler to make some physical types fail when overriding `ordering`.

Reply:
I would say it is easier to use with pattern matching and to have proper error management in place (instead of try..catch...).

Reply:
do we really need try-catch? It seems OK to always throw `QueryExecutionErrors.orderedOperationUnsupportedByDataTypeError`

Reply:
Ok, works for me.